### PR TITLE
Bug 1877281: [vSphere] Don't fail when tag is not found

### PR DESCRIPTION
**Pull request by alexander-demichev**
When a cluster that was first created via UPI the clusterID tag is missed in the vCenter.
This causes the machine controller to create the instances but failing and breaking the loop before being able to reconcile the status appropriately resulting in the machine resource having an inconsistent status which does not reflect reality.

`github.com/vmware/govmomi` misses typed errors, so the only way to not fail reconciliation on not found tag is iterate through the list of all tags and try to find clusterID tag. We assume the tag is always present in IPI.

### CODE REVIEW

**Comment by JoelSpeed** [comment]
Is there a BZ related to this?

Can you please ensure that there are unit tests to cover this at a minimum

**Comment by alexander-demichev** [comment]
We don't have a BZ, it's a jira ticket

**Comment by alexander-demichev** [comment]
@michaelgugino What you say is true, do we require cluster tags for other providers' UPI?  Do you if that's documented somewhere?

**Comment by michaelgugino** [comment]
> @michaelgugino What you say is true, do we require cluster tags for other providers' UPI? Do you know if that's documented somewhere?

I skimmed through the vSphere UPI docs, I didn't see anything about tags.

I am not sure how everything works in this provider.  If we can able to query machines from vsphere on a unique basis (eg, every VM has a unique name that we know)

File: pkg/controller/vsphere/reconciler.go [hunk 1]
```diff
@@ -882,6 +882,16 @@ func (vm *virtualMachine) reconcileTags(ctx context.Context, session *session.Se
 
 // checkAttachedTag returns true if tag is already attached to a vm
```
Comment (enxebre) [inline pkg/controller/vsphere/reconciler.go:883]: the comment needs to be updated

**Comment by enxebre** [comment]
In vSphere we use the s.machine.UID or fallback to the s.machine.Name to find instances. As far as I'm aware the clusterID tag is only used by the installer to remove resources on a destroy command so this change seems safe. This would let UPI born clusters to work smoothly.

I also agree with this though https://github.com/openshift/machine-api-operator/pull/667#pullrequestreview-465915278
https://bugzilla.redhat.com/show_bug.cgi?id=1877281

**Comment by enxebre** [comment]
/retitle Bug 1877281: [vSphere] Don't fail when tag is not found

**Review by JoelSpeed — COMMENTED** [review]
/lgtm

**Pull request merged**